core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -243,11 +240,8 @@ class JDBCSourceAsColumnarStore(private var _connProperties: ConnectionPropertie
   def closeConnection(c: Option[Connection]): Unit = {
     c match {
       case Some(conn) if !conn.isClosed =>
-        conn match {
-          case ConnectionType.Embedded =>
-            // conn commit removed txState from the conn context.
-            conn.commit()
-            conn.close()
+        connectionType match {
+          case ConnectionType.Embedded => // Do nothing
```

Comment:
With this, I get this error:
java.sql.BatchUpdateException: (Server=localhost/127.0.0.1[20997] Thread=pool-6-thread-5) Cannot issue commit in a nested connection when there is a pending operation for 'insert into order_line_col values(6, 6, '6')' in the parent connection.
	at io.snappydata.thrift.SnappyDataService$executeUpdate_result$executeUpdate_resultStandardScheme.read(SnappyDataService.java:8244)
	at io.snappydata.thrift.SnappyDataService$executeUpdate_result$executeUpdate_resultStandardScheme.read(SnappyDataService.java:8221)
	at io.snappydata.thrift.SnappyDataService$executeUpdate_result.read(SnappyDataService.java:8160)
	at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:86)
	at io.snappydata.thrift.SnappyDataService$Client.recv_executeUpdate(SnappyDataService.java:285)
	at io.snappydata.thrift.SnappyDataService$Client.executeUpdate(SnappyDataService.java:269)
	at io.snappydata.thrift.internal.ClientService.executeUpdate(ClientService.java:933)
	at io.snappydata.thrift.internal.ClientStatement.executeBatch(ClientStatement.java:518)
	at io.snappydata.cluster.QueryRoutingDUnitSecurityTest$$anonfun$batchInsert$1.apply$mcVI$sp(QueryRoutingDUnitSecurityTest.scala:257)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:160)
	at io.snappydata.cluster.QueryRoutingDUnitSecurityTest$.batchInsert(QueryRoutingDUnitSecurityTest.scala:253)
	at io.snappydata.cluster.QueryRoutingDUnitSecurityTest$.columnTableRouting(QueryRoutingDUnitSecurityTest.scala:85)
	at io.snappydata.cluster.QueryRoutingDUnitSecurityTest.testColumnTableRouting(QueryRoutingDUnitSecurityTest.scala:42)